### PR TITLE
r/migration: initial support for lint

### PR DESCRIPTION
**Pull request by giautm**
Signed-off-by: Giau. Tran Minh <hello@giautm.dev>

### CODE REVIEW

File: internal/atlas/atlas.go [hunk 1]
```diff
@@ -33,6 +33,13 @@ type (
 		URL             string
 		RevisionsSchema string
 	}
+	// LintParams are the parameters for the `migrate lint` command.
+	LintParams struct {
+		DevURL  string
+		DirURL  string
+		Latest  uint
+		GitBase bool
```
Comment (a8m) [inline internal/atlas/atlas.go:41]: GitBase is not a real option, no?
```suggestion
		Latest  uint
```

File: internal/atlas/atlas.go [hunk 2]
```diff
@@ -77,6 +84,26 @@ func (c *Client) Apply(ctx context.Context, data *ApplyParams) (*ApplyReport, er
...
 
+// Lint runs the `migrate lint` command.
+func (c *Client) Lint(ctx context.Context, data *LintParams) (*SummaryReport, error) {
+	args := []string{
+		"migrate", "lint", "--log", "{{ json . }}",
+		"--dev-url", data.DevURL,
+		"--dir", fmt.Sprintf("file://%s", data.DirURL),
+	}
+	if data.Latest > 0 {
+		args = append(args, "--latest", fmt.Sprintf("%d", data.Latest))
```
Comment (a8m) [inline internal/atlas/atlas.go:95]: https://pkg.go.dev/strconv#FormatUint

File: internal/atlas/atlas.go (hunk 1 above)
Comment (giautm) [reply internal/atlas/atlas.go:41]: Yes, we don't use it yet

File: internal/provider/atlas_migration_resource_test.go [hunk 3]
```diff
@@ -158,6 +160,48 @@ func TestAccMigrationResource(t *testing.T) {
...
+				}`,
+					fmt.Sprintf("%s/%s", mysqlURL, schema4),
+					fmt.Sprintf("%s/%s", mysqlDevURL, schema4),
+				),
+				ExpectError: regexp.MustCompile("You have an error in your SQL syntax"),
+			},
+		},
+	})
+
+	// // Lint-Drop
```
Comment (a8m) [inline internal/provider/atlas_migration_resource_test.go:185]: ?

**Review by a8m — APPROVED** [review]
🚀

**Pull request merged**